sdk/containerregistry/azure-containerregistry/azure/containerregistry/_models.py

```diff
@@ -211,53 +266,82 @@ def _to_generated(self) -> RepositoryWriteableProperties:
             can_list=self.can_list,
         )
 
-    def __getattr__(self, name):
+    def __getattr__(self, name: str) -> datetime:
         if name == "last_udpated_on":
             warnings.warn(
-                "The property name with a typo called 'last_udpated_on' has been deprecated and will be retired in future versions",  # pylint: disable=line-too-long
+                "The property name with a typo called 'last_udpated_on' has been deprecated and will be retired \
+                in future versions",
                 DeprecationWarning,
             )
-            return self.last_updated_on
-        return super().__getattr__(self, name)  # pylint: disable=no-member
+        return self.last_updated_on
 
     @property
     def created_on(self) -> datetime:
+        """Time and date the repository was created.
+
+        :rtype: ~datetime.datetime
+        """
         return self._created_on
 
     @property
     def last_updated_on(self) -> datetime:
+        """Time and date the repository was last updated.
+
+        :rtype: ~datetime.datetime
+        """
         return self._last_updated_on
 
     @property
     def manifest_count(self) -> int:
+        """Number of manifest in the repository.
+
+        :rtype: int
+        """
         return self._manifest_count
 
     @property
     def name(self) -> str:
+        """Name of the repository.
+
+        :rtype: str
+        """
         return self._name
 
     @property
     def tag_count(self) -> int:
+        """Number of tags associated with the repository.
+
+        :rtype: int
+        """
         return self._tag_count
 
 
 class ArtifactTagProperties:
-    """Represents properties of a single tag
+    """Represents properties of a single tag.
 
-    :ivar bool can_delete: Delete Permissions for a tag.
-    :ivar bool can_write: Write Permissions for a tag.
-    :ivar bool can_read: Read Permissions for a tag.
-    :ivar bool can_list: List Permissions for a tag.
+    :ivar Optional[bool] can_delete: Delete Permissions for a tag.
```

Comment:
I don't think we been _both_ `:ivar` docstrings as well as the documented classattributes on lines 335+.
Doesn't this create duplicate documentation?

Reply:
After checking the latest generated doc, yes, it does create duplicate documentation. We should keep either one, so I have removed the class attributes and it's documentation on `can_XXX`.